src/application/worker/index.worker.js

```diff
@@ -9,42 +9,105 @@ async function start() {
   const store = require("./store").default;
   const loop = require("./loop").default;
   const grabCanvasPlugin = require("../plugins/grab-canvas").default;
+  const get = require("lodash.get");
 
   const { tick: frameTick } = require("./frame-counter");
   const { getFeatures, setFeatures } = require("./audio-features");
   // const featureAssignmentPlugin = require("../plugins/feature-assignment");
 
   let interval = store.getters["fps/interval"];
 
+  const commitQueue = [];
+
   store.subscribe(mutation => {
-    const { type, payload } = mutation;
+    const { type: mutationType, payload: mutationPayload } = mutation;
 
-    if (type === "beats/SET_BPM" || type === "fps/SET_FPS") {
-      store.dispatch("tweens/updateBpm", { bpm: payload.bpm });
+    if (mutationType === "beats/SET_BPM" || mutationType === "fps/SET_FPS") {
+      store.dispatch("tweens/updateBpm", { bpm: mutationPayload.bpm });
     }
 
-    if (type === "beats/SET_KICK" && payload.kick === lastKick) {
+    if (
+      mutationType === "beats/SET_KICK" &&
+      mutationPayload.kick === lastKick
+    ) {
       return;
-    } else if (type === "beats/SET_KICK" && payload.kick !== lastKick) {
-      lastKick = payload.kick;
+    } else if (
+      mutationType === "beats/SET_KICK" &&
+      mutationPayload.kick !== lastKick
+    ) {
+      lastKick = mutationPayload.kick;
     }
 
-    if (type === "fps/SET_FPS") {
+    if (mutationType === "fps/SET_FPS") {
       interval = store.getters["fps/interval"];
     }
 
     if (
-      type === "modules/UPDATE_ACTIVE_MODULE" &&
-      (payload.key !== "props" || payload.key !== "meta")
+      mutationType === "modules/UPDATE_ACTIVE_MODULE" &&
+      (mutationPayload.key !== "props" || mutationPayload.key !== "meta")
     ) {
       return;
     }
 
+    const {
+      inputs: { inputs, inputLinks }
+    } = store.state;
+
+    // Update mutation type Input Links
+    const mutationTypeInputLinks = Object.values(inputLinks).filter(
+      link => link.type === "mutation"
+    );
+    const inputLinksLength = mutationTypeInputLinks.length;
+    for (let i = 0; i < inputLinksLength; ++i) {
+      const link = mutationTypeInputLinks[i];
+      const inputId = link.id;
+      const bind = inputs[inputId];
+
+      const { type, location, data } = bind;
+
+      const { type: linkType, location: linkLocation, match } = link;
+
+      if (linkType !== "mutation" || match.type !== mutationType) {
+        continue;
+      }
+
+      let payloadMatches = false;
+
+      if (match.payload) {
+        const matchPayloadKeys = Object.keys(match.payload);
+        payloadMatches = matchPayloadKeys.every(key => {
+          const value = match.payload[key];
+          return value === mutationPayload[key];
+        });
+      } else {
+        payloadMatches = true;
+      }
+
+      if (!payloadMatches) {
```

Comment:
If we have no payloadMatch, we continue? Why do we check if the payload is matching anyway? To see if there was a change?

Reply:
The payload needs to match to check for things like IDs or other unique parameters so we know the correct inputLink will be updated.

If we didn't check for the right ID on the MIDI input links, all MIDI mutations would update on any MIDI input which would be wasteful in terms of postMessage.